numpy/distutils/fcompiler/__init__.py

```diff
@@ -681,6 +684,14 @@ def link(self, target_desc, objects,
             except DistutilsExecError as e:
                 msg = str(e)
                 raise LinkError(msg)
+            if sys.platform == 'cygwin':
+                # Rebase newly-compiled module so it has a higher
+                # chance of surviving fork()
+                COMPILED_MODULES.append(output_filename)
+                rebase_args = ['/usr/bin/rebase', '--database',
+                               '--oblivious']
+                rebase_args.extend(COMPILED_MODULES)
+                self.spawn(rebase_args)
```

Comment:
I'm going to see if I can find it, but at some point I wrote (for Sage) some code similar to this but a little more advanced which tried to find an open memory range to rebase compiled modules to.  I don't recall why exactly but just using `rebase` directly wasn't always working for me (sometimes the rebase database wasn't always available).  I don't recall the details though.

Of course, in general this is the right idea.

Reply:
The tests run better without this than with it at the moment.  I think the problem is not including the numpy dlls themselves in the `rebase`.  There is a way to list those using pip, which would allow be to include those in the list passed to `rebase`, but it doesn't look like that's needed on 64-bit Cygwin at the moment.  I'll remove this code soon.

Reply:
I removed this code.